solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -0,0 +1,1386 @@
+// SPDX-License-Identifier: SEE LICENSE IN LICENSE
+pragma solidity 0.6.12;
+import "../../ConverterVersion.sol";
+import "../../interfaces/IConverter.sol";
+import "../../interfaces/IConverterAnchor.sol";
+import "../../interfaces/IConverterUpgrader.sol";
+import "../../../token/interfaces/IDSToken.sol";
+import "../../../utility/ContractRegistryClient.sol";
+import "../../../utility/ReentrancyGuard.sol";
+import "../../../utility/SafeMath.sol";
+import "../../../utility/TokenHandler.sol";
+import "../../../utility/TokenHolder.sol";
+import "../../../utility/Math.sol";
+
+/**
+ * @dev This contract is a specialized version of the converter, which is
+ * optimized for a liquidity pool that has 2 reserves with 50%/50% weights.
+ */
+contract StandardPoolConverter is
+    ConverterVersion,
+    IConverter,
+    TokenHandler,
+    TokenHolder,
+    ContractRegistryClient,
+    ReentrancyGuard
+{
+    using SafeMath for uint256;
+    using Math for *;
+
+    IERC20Token private constant ETH_RESERVE_ADDRESS = IERC20Token(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
+    uint256 private constant MAX_UINT128 = 2**128 - 1;
+    uint256 private constant MAX_UINT112 = 2**112 - 1;
+    uint256 private constant AVERAGE_RATE_PERIOD = 10 minutes;
+    uint32 private constant PPM_RESOLUTION = 1000000;
+
+    uint256 private __reserveBalances;
+    IERC20Token[] private __reserveTokens;
+    mapping(IERC20Token => uint256) private __reserveIds;
+
+    IConverterAnchor public override anchor; // converter anchor contract
+    uint32 public override maxConversionFee; // maximum conversion fee, represented in ppm, 0...1000000
+    uint32 public override conversionFee; // current conversion fee, represented in ppm, 0...maxConversionFee
+
+    // average rate details:
+    // bits 0...111 represent the numerator of the rate between reserve token 0 and reserve token 1
+    // bits 111...223 represent the denominator of the rate between reserve token 0 and reserve token 1
+    // bits 224...255 represent the update-time of the rate between reserve token 0 and reserve token 1
+    // where `numerator / denominator` gives the worth of one reserve token 0 in units of reserve token 1
+    uint256 public averageRateInfo;
+
+    /**
+     * @dev triggered when the converter is activated
+     *
+     * @param _type        converter type
+     * @param _anchor      converter anchor
+     * @param _activated   true if the converter was activated, false if it was deactivated
+     */
+    event Activation(uint16 indexed _type, IConverterAnchor indexed _anchor, bool indexed _activated);
```

Comment:
Since you don't inherit from `ConverterBase`, you can still move shared definitions to `IConverter.sol` (for example)

Reply:
Like many of these events

Reply:
@yudilevi : Move all events from `ConverterBase` to `IConverter`?

Reply:
@barakman sure

Reply:
Done